dev/io.openliberty.jakartaee10.internal_fat/fat/src/io/openliberty/jakartaee10/internal/apps/jakartaee10/web/WebProfile10TestServlet.java

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.jakartaee10.internal.apps.jakartaee10.web;
+
+import static org.junit.Assert.assertEquals;
+
+import java.sql.Connection;
+
+import javax.naming.InitialContext;
+import javax.sql.DataSource;
+
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+import io.openliberty.jakartaee10.internal.apps.jakartaee10.web.cdi.BasicCDIBean;
+import jakarta.annotation.Resource;
+import jakarta.inject.Inject;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.transaction.UserTransaction;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+
+@SuppressWarnings("serial")
+@WebServlet("/WebProfile10TestServlet")
+public class WebProfile10TestServlet extends FATServlet {
+
+    @Inject
+    BasicCDIBean basicBean;
+
+//    @PersistenceContext(unitName = "JPAPU")
+//    private EntityManager em;
+
+    @Resource
+    private UserTransaction tx;
+
+    @Test
+    public void testCDI() throws Exception {
+        assertEquals(BasicCDIBean.MSG, basicBean.sayHi());
+    }
+
+    @Test
+    public void testJAXRS(HttpServletRequest req, HttpServletResponse resp) throws Exception {
+        Client client = ClientBuilder.newBuilder().build();
+        try {
+            String url = "http://" + req.getServerName() + ':' + req.getServerPort() + "/webProfile10App/rest/test";
+            System.out.println("Making a PATCH request to URL: " + url);
+            String result = client.target(url)
+                            .request()
+                            .method("PATCH", String.class);
+            assertEquals("patch-success", result);
+        } finally {
+            client.close();
+        }
+    }
+
+    //TODO these three tests should be enabled when JPA and JSONB are ready
```

Comment:
Are they not ready?  We have dropped in code for both.

Reply:
the tests didn't work out of the box ... i didn't investigate very hard though

Reply:
Was able to re-enable the JPA test but the JSONB ones still fail with a J2S issue.